src/modules/core/hooks/useGetTokenInfos.ts

```diff
@@ -25,4 +32,24 @@ const useGetTokenInfos = ({
   return { assetsInfo, assetAmount };
 };
 
-export { useGetTokenInfos };
+const useGetNftsInfos = ({ assetId, nftList }: IUseGetNftInfos) => {
+  const nftsInfo = useMemo(() => {
+    return nftList.find((nft) => nft.assetId === assetId);
+  }, [assetId, nftList]);
+
+  let nftImageUrl = null;
+
+  if (nftsInfo?.metadata) {
+    const imageKeys = ['image'];
+    const imageKey = Object.keys(nftsInfo.metadata).find((key) =>
+      imageKeys.includes(key.split(':').at(0)!),
+    );
+    const nftnftImageUrl = parseURI(nftsInfo.metadata[imageKey!]);
+    const metadata = nftsInfo.metadata as defaultMetadata;
+    nftImageUrl = nftnftImageUrl || metadata.image;
+  }
```

Comment:
```suggestion
   const nftImageUrl = useMemo(() => {
    if (!nftsInfo?.metadata) return null;
    
    const imageKeys = ['image'];
    const imageKey = Object.keys(nftsInfo.metadata).find(
      (key) => imageKeys.includes(key.split(':')[0])
    );
    
    if (imageKey) {
      const parsedUrl = parseURI(nftsInfo.metadata[imageKey]);
      if (parsedUrl) return parsedUrl;
    }
    
    return (nftsInfo.metadata as defaultMetadata).image || null;
  }, [nftsInfo]);
```

add an `useMemo` in image to avoid unnecessary recomputation